cloud/parameterstore/ssm_client.go

```diff
@@ -0,0 +1,178 @@
+	DeleteParameters(context.Context, *ssm.DeleteParametersInput) ([]*ssm.DeleteParametersOutput, error)
+	// GetParametersSimple is the same as GetParameters but only returns the
+	// parameter information rather than the full output.
+	GetParametersSimple(context.Context, *ssm.GetParametersInput) ([]ssmTypes.Parameter, error)
+	// GetParameters retrieves the specified parameters. Parameter Store
+	// limits how many parameters can be retrieved per call, so implementations
+	// must handle batching transparently. The returned output slice contains
+	// the output of each batched request.
+	GetParameters(context.Context, *ssm.GetParametersInput) ([]*ssm.GetParametersOutput, error)
+}
+
+type ssmClientImpl struct {
+	client *ssm.Client
+}
+
+// configCache is a mapping of configuration options to the cached AWS
+// configuration created by it. Currently the only supported option is the AWS
+// region.
+var configCache map[string]*aws.Config = make(map[string]*aws.Config)
+
+func newSSMClient(ctx context.Context, region string) (ssmClient, error) {
+	if region == "" {
+		region = "us-east-1"
+	}
+
+	if configCache[region] == nil {
+		cfg, err := config.LoadDefaultConfig(ctx,
+			config.WithRegion(region),
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "loading AWS config")
+		}
+		otelaws.AppendMiddlewares(&cfg.APIOptions)
+		configCache[region] = &cfg
+	}
+
+	c := ssm.NewFromConfig(*configCache[region])
+
+	return &ssmClientImpl{client: c}, nil
+}
+
+func (c *ssmClientImpl) PutParameter(ctx context.Context, input *ssm.PutParameterInput) (*ssm.PutParameterOutput, error) {
+	return retrySSMClientOp(ctx, "PutParameter", c.client, input, func(ctx context.Context, client *ssm.Client, input *ssm.PutParameterInput) (*ssm.PutParameterOutput, error) {
+		return client.PutParameter(ctx, input)
+	})
+}
+
+func (c *ssmClientImpl) DeleteParametersSimple(ctx context.Context, input *ssm.DeleteParametersInput) ([]string, error) {
+	output, err := c.DeleteParameters(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	var deletedParams []string
+	for _, o := range output {
+		deletedParams = append(deletedParams, o.DeletedParameters...)
+	}
+	return deletedParams, nil
+}
+
+func (c *ssmClientImpl) DeleteParameters(ctx context.Context, input *ssm.DeleteParametersInput) ([]*ssm.DeleteParametersOutput, error) {
+	// TODO (DEVPROD-9391): follow-up PR should handle batching requests due
+	// to limit of 10 parameters per API call.
+	output, err := retrySSMClientOp(ctx, "DeleteParameters", c.client, input, func(ctx context.Context, client *ssm.Client, input *ssm.DeleteParametersInput) (*ssm.DeleteParametersOutput, error) {
+		return client.DeleteParameters(ctx, input)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return []*ssm.DeleteParametersOutput{output}, nil
+}
+
+func (c *ssmClientImpl) GetParametersSimple(ctx context.Context, input *ssm.GetParametersInput) ([]ssmTypes.Parameter, error) {
+	allOutputs, err := c.GetParameters(ctx, input)
```

Comment:
same retrySSMClientOp comment

Reply:
Same thing as above, this is just a wrapper for `GetParameters` to simplify down the output result to the thing that a caller is most likely to care about.